src/server/page_server.cpp

```diff
@@ -44,80 +44,56 @@ page_server::~page_server ()
   assert (m_active_tran_server_conn == nullptr);
 }
 
-void page_server::set_active_tran_server_connection (cubcomm::channel &&chn)
+page_server::connection_handler::connection_handler (cubcomm::channel &chn, page_server *ps)
```

Comment:
ps can never be null. Use a reference instead of pointer to force the argument be passed an actual instance.
```suggestion
page_server::connection_handler::connection_handler (cubcomm::channel &chn, page_server &ps)
```